Zend/zend_alloc.c

```diff
@@ -1047,27 +1178,31 @@ static void *zend_mm_alloc_pages(zend_mm_heap *heap, uint32_t pages_count ZEND_F
 get_chunk:
 			if (heap->cached_chunks) {
 				heap->cached_chunks_count--;
+				ZEND_MM_POISON_CHUNK_HDR(chunk, heap);
 				chunk = heap->cached_chunks;
+				ZEND_MM_UNPOISON_CHUNK_HDR(chunk);
```

Comment:
Isn't this placed a line too early?

Reply:
It's used in the next line in chunk->next, seems correct to me